src/plugins/topic_viewer/TopicViewer_TEST.cc

```diff
@@ -0,0 +1,146 @@
+
+    EXPECT_TRUE(app.LoadPlugin("TopicViewer"));
+
+    // Get main window
+    auto win = app.findChild<MainWindow *>();
+    ASSERT_NE(nullptr, win);
+
+    // Get plugin
+    auto plugins = win->findChildren<Plugin *>();
+    EXPECT_EQ(plugins.size(), 1);
+
+    auto plugin = plugins[0];
+    EXPECT_EQ(plugin->Title(), "Topic Viewer");
+
+    // Cleanup
+    plugins.clear();
+}
+
+TEST(TopicViewerTest, Model)
+{
+    // =========== Publish =================
+    transport::Node node;
+
+    // int
+    auto pubInt = node.Advertise<msgs::Int32>("/IntTopic");
+    msgs::Int32 msgInt;
+
+    // collision
+    auto pub = node.Advertise<msgs::Collision> ("/CollisionTopic");
+    msgs::Collision msg;
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(300));
+
+    pub.Publish(msg);
+    pubInt.Publish(msgInt);
+
+    // ========== Load the Plugin ============
+    common::Console::SetVerbosity(4);
+
+    Application app(g_argc, g_argv);
+    app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
+
+    EXPECT_TRUE(app.LoadPlugin("TopicViewer"));
+
+    // Get main window
+    auto win = app.findChild<MainWindow *>();
+    EXPECT_NE(nullptr, win);
+
+    // Get plugin
+    auto plugins = win->findChildren<plugins::TopicViewer *>();
+    ASSERT_EQ(plugins.size(), 1);
+
+    auto plugin = plugins[0];
+    ASSERT_NE(nullptr, plugin);
+
+    // ============= Model =====================
+    auto model = plugin->Model();
+    ASSERT_NE(model, nullptr);
+
+    auto root = model->invisibleRootItem();
+    ASSERT_NE(model, nullptr);
+
+    ASSERT_EQ(root->hasChildren(), true);
+
+    bool foundCollision = false;
+    bool foundInt = false;
+
+    for (int i = 0; i < root->rowCount(); ++i)
+    {
+        auto child = root->child(i);
+
+        if (child->data(NAME_ROLE) == "/CollisionTopic")
+        {
+            foundCollision = true;
+
+            EXPECT_EQ(child->data(TYPE_ROLE), "ignition.msgs.Collision");
+            EXPECT_EQ(child->rowCount(), 8);
+
+            auto pose = child->child(5);
+            auto position = pose->child(3);
+            auto x = position->child(1);
+
+            EXPECT_EQ(x->data(NAME_ROLE), "x");
+            EXPECT_EQ(x->data(TYPE_ROLE), "double");
+            EXPECT_EQ(x->data(PATH_ROLE), "pose-position-x");
+            EXPECT_EQ(x->data(TOPIC_ROLE), "/CollisionTopic");
+            EXPECT_EQ(x->data(PLOT_ROLE), true);
+        }
+        else if (child->data(NAME_ROLE) == "/IntTopic")
+        {
+            foundInt = true;
+
+            EXPECT_EQ(child->data(TYPE_ROLE), "ignition.msgs.Int32");
+            EXPECT_EQ(child->rowCount(), 2);
+
+            auto data = child->child(1);
+
+            EXPECT_EQ(data->data(NAME_ROLE), "data");
+            EXPECT_EQ(data->data(TYPE_ROLE), "int32");
+            EXPECT_EQ(data->data(PATH_ROLE), "data");
+            EXPECT_EQ(data->data(TOPIC_ROLE), "/IntTopic");
+            EXPECT_EQ(data->data(PLOT_ROLE), true);
+        }
```

Comment:
How about adding something like this, so we're sure no other topics are listed?

```
else
{
  EXPECT_TRUE(false);
}
```

Reply:
but the plugin list the topics that is currently in the transport 
so this will make the test case fails if we publishing anything to the transport when performing the testing
for example, if you run the test while you are opening ign-gazebo this will make it fails

Reply:
Ahh yes, that makes sense. One way to work around this is to set a different partition in the beginning of the test, so that all nodes starting during the test can only see other nodes also in the test. Could you give this a try for example?

`setenv("IGN_PARTITION", "ign-gazebo-test", 1);`